Edits to our Graph Platform article to highlight the supergraph (#137)
apollographql/docs: src/content/basics/intro/platform.mdx

---
title: The Apollo Supergraph Platform
description: Bring your supergraph from prototype to production
---

Apollo is a platform for building, managing, and scaling a **unified supergraph**: a data layer that combines your organization's backend services into a single, all-purpose API. With just one query to your supergraph, an application client can fetch data from any combination of sources:

```mermaid
graph LR;
  clients([Client]);
  subgraph "Your supergraph";
  gateway([Graph<br/>router]);
  serviceA[Users<br/>subgraph];
  serviceB[Products<br/>subgraph];
  dbA[(Users<br/>DB)];
  dbB[(Products<br/>DB)];
  gateway <--- serviceA & serviceB;
  serviceA --- dbA;
  serviceB --- dbB;
  end;
  clients -.- gateway;
  class clients secondary;
```

> Clients query your supergraph with a powerful language called [GraphQL](./benefits/), which enables them to retrieve exactly the data they need with no overfetching. The architecture of your supergraph is not exposed, and clients query it exactly as though it were a "monolithic" GraphQL server.

Reviewer: ```suggestion
Clients query your supergraph with a powerful language called [GraphQL](./benefits/), which enables them to retrieve exactly the data they need with no overfetching. The architecture of your supergraph is not exposed, and clients query it from a single endpoint.
```

* One or more **subgraphs**
    * Each subgraph is itself a GraphQL API that's responsible for populating a subset of your supergraph's available data.
    * Usually, each of your existing backend services acts as a different subgraph.
* Backing **data sources** (databases, REST APIs, etc.)
    * Your subgraphs use these to populate data for incoming queries.

The diagram at the top of this page shows a basic supergraph with two subgraphs and two corresponding data sources.

The following video provides a helpful visualization of how the Apollo Federation architecture works:

<YouTube youTubeId="wRExDdgs6JU" />

## What are the benefits of a supergraph?

Adopting a supergraph significantly improves the developer experience for every team (both internal and external) that interacts with your organization's data. These improvements enable developers to ship products and features more quickly, with greater confidence in their reliability and performance.

* Many benefits of a supergraph are those provided by the GraphQL language and the Apollo Federation architecture.
    * [GraphQL benefits](./benefits/)
    * [Federation benefits](/federation/#benefits-of-federation)

> * A unified supergraph enables the use of powerful tooling for schema management, metrics aggregation, and change validation. [Learn about Apollo Studio.](#how-do-i-manage-my-supergraph)

Reviewer: ```suggestion
* A supergraph enables the use of powerful tooling for schema management, metrics aggregation, and change validation. [Learn about Apollo Studio.](#how-do-i-manage-my-supergraph)
```
